pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -482,7 +506,20 @@ class CollectionUploadViewSet(
     serializer_class = CollectionVersionUploadSerializer
     pulp_tag_name = "Pulp_Ansible: Artifacts Collections V3"
 
-    DEFAULT_ACCESS_POLICY = _PERMISSIVE_ACCESS_POLICY
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            _CAN_VIEW_REPO_CONTENT,
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_perms:ansible.add_collection",
+                    "has_model_or_obj_perms:ansible.view_ansiblerepository",
```

Comment:
I'm not sure this is doing what you think it is doing. `has_model_or_obj_perms` calls the viewset's `get_object` method if the user doesn't have model perms, and this viewsets `get_object` method behavior is undefined (possibly will throw an error) because `queryset = None`. I think this condition should use the new one you wrote, `v3_can_view_repo_content`.